server/RemoteHubClient.ts

```diff
@@ -190,4 +202,60 @@ export default class RemoteHubClient extends BaseRemoteClient {
       else callback(null);
     });
   };
+
+  private onImportProject = (data: ImportProjectData) => {
+    // Import a project with a zip archive
+    // 1] Extract the content of the zip file in a temporary folder
+    // 2] Move the extracted content into the projects' folder
+
+    // Extract the content of the zip file in a temporary folder
+    temp.mkdir("project", (err: Error, dirPath: any) => {
+      const inputPath = path.join(dirPath, "project.zip");
+      let rootFolderName: string = null;
+      fs.writeFile(inputPath, data.data, (err) => {
+        if (err) throw err;
+
+        yauzl.open(inputPath, { lazyEntries: true }, (err: Error, zipFile: any) => {
+          if (err != null) throw err;
+
+          zipFile.readEntry();
+          zipFile.on("entry", (entry: any) => {
+            if (rootFolderName === null) {
+              rootFolderName = entry.fileName;
+            }
+
+            if (/\/$/.test(entry.fileName)) {
+              mkdirp(path.join(dirPath, entry.fileName), (err: Error) => {
+                if (err != null) throw err;
+                zipFile.readEntry();
+              });
+            }
+            else {
+              zipFile.openReadStream(entry, (err: Error, readStream: NodeJS.ReadableStream) => {
+                if (err != null) throw err;
+
+                mkdirp(path.dirname(path.join(dirPath, entry.fileName)), (err: Error) => {
+                  if (err != null) throw err;
+
+                  readStream.pipe(fs.createWriteStream(path.join(dirPath, entry.fileName)));
+                  readStream.on("end", () => {
+                    zipFile.readEntry();
+                  });
+                });
+              });
+            }
+          });
+          zipFile.on("end", () => {
+            // move the extracted content into the projects' folder
+            copydir(path.join(dirPath, rootFolderName), path.join(this.server.projectsPath, rootFolderName), (err: Error) => {
```

Comment:
Well, when I tried with fs.rename I had a crash on the server side with on error like this:

`Error: EXDEV: cross-device link not permitted, rename '/tmp/project1161019-14225-1cpphb6/dodgell-master/' -> '/home/ilphrin/Jeux/superpowers/core/projects/dodgell-master/'`

I didn't find a solution to this, so I tried copying the folder and then it worked